### PR TITLE
Add version attribute to force fields

### LINKED ISSUE #187: Maintaining different versions of force fields
Not sure we've talked about this much, but I think it's worth recording somewhere.

Should version control be handled within `foyer`, in any capacity, or should it be done purely externally, i.e. with git?

Hypothetically (read: something I need done for my own research) I have a force field with recently-updated parameters, but it is useful to have an old version for comparisons. It happens to be buried in the git history, but it's not so easy to grab an old version of a force field just using git. It's not like we have `myff = Forcefield('myff.xml', commit='blah123')`. Grabbing an older version of the file and using just using `myff_old.xml` would work for my purposes, but it doesn't seem to be preferred.

### PR DESCRIPTION
**Pull request by mattwthompson**
### PR Summary:

Fixes  #187, would be useful in  #290 

* Adds `version` as an attribute of the `ForceField` element in the XML schema
* Adds `version` as an attribute to the `Forcefield` class
* Adds a simple test via reading a toy XML file and asserting the above two `version`s are equal

This is not done elegantly right now; 99% of the XML loading is done by OpenMM, so I had to do this after the fact and only dealing with the case of passing a single XML, which is all I believe we do right now. I'm not sure what the target for this PR should be, but I think we should soon have a bigger discussion about how we want to handle multiple XML files. The OpenMM use case is basically `Forcefield('protein_model.xml', 'water_model.xml')`, i.e. there is a clean interface between constituents and not any expected overlap between the two. This doesn't cleanly map on to our use cases, I don't think. Related to #297 #262 #149

If this seems reasonable I can do the same for names (#153), here or in another PR. Requesting a few specific reviewers since I have discussed this with many people over the past year or two (but only gotten to it now 🤕)

### PR Checklist
------------
 - [x] Includes appropriate unit test(s)
 - [ ] Appropriate docstring(s) are added/updated
 - [x] Code is (approximately) PEP8 compliant
 - [x] Issue(s) raised/addressed?

### CODE REVIEW

**Comment by ahy3nz** [comment]
```
 /Users/ayang41/Programs/foyer/foyer/tests/test_forcefield.py(511)test_load_version_number()
-> assert lj_ff.version == '0.4.1'
(Pdb) l
506  	            assert str(item.xpath('comment()')) == '[<!--Note: original overrides="opls_144"-->]'
507
508  	def test_load_version_number():
509  	    lj_ff = Forcefield(get_fn('lj.xml'))
510  	    import pdb; pdb.set_trace()
511  ->	    assert lj_ff.version == '0.4.1'
[EOF]
(Pdb) foo = Forcefield(forcefield_files=[get_fn('lj.xml')])
(Pdb) foo.version
'foobar'
(Pdb) lj_ff.version
'0.4.1'
(Pdb)
```

* You have an extra `foobar` debug line in the init
* Did you want to cover the case where someone passes a list of XML files? This PR appears to only cover the case of passing a single XML string, not a list (which is what most people tend to do anyways...)

**Comment by mattwthompson** [comment]
> Did you want to cover the case where someone passes a list of XML files? This PR appears to only cover the case of passing a single XML string, not a list (which is what most people tend to do anyways...)

I should be able to rework it to handle multiple files but I also think this might be a feature we want to drop at some point.

**Comment by mattwthompson** [comment]
I'm happy with where this is at now. If this approach is ok, I will duplicate this functionality to support a `name` attribute.

File: foyer/forcefield.py [hunk 1]
```diff
@@ -385,6 +403,24 @@ def included_forcefields(self):
             self._included_forcefields[basename] = ff_filepath
         return self._included_forcefields
 
+    def _parse_version_number(self, forcefield_file):
+        with open(forcefield_file, 'r') as f:
+            tree = ET.parse(f)
+            root = tree.getroot()
+            try:
+                return root.attrib['version']
+            except KeyError:
```
Comment (justinGilmer) [inline foyer/forcefield.py]: Going off of Mike's comment about raising a warning, this seems like the place where that would go.
```suggestion
            except KeyError:
                warn.Warn("version warning message here")
                return None
```

**Pull request merged**